models/health.go

```diff
@@ -70,6 +70,26 @@ type WorkloadStatus struct {
 	DesiredReplicas   int32  `json:"desiredReplicas"`
 	CurrentReplicas   int32  `json:"currentReplicas"`
 	AvailableReplicas int32  `json:"availableReplicas"`
+	SyncedProxies     int32  `json:"syncedProxies"`
+}
+
+type ProxyStatuses string
+
+const (
+	Synced  ProxyStatuses = "Synced"
+	NotSent ProxyStatuses = "NOT_SENT"
+	Stale   ProxyStatuses = "Stale"
+	StaleNa ProxyStatuses = "Stale (Never Acknowledged)"
+)
+
+// ProxyStatus gives the sync status of the sidecar proxy.
+// In healthy scenarios all variables should be true.
+// If at least one variable is false, then the proxy isn't fully sync'ed with pilot.
+type ProxyStatus struct {
+	CDS ProxyStatuses `json:"CDS"`
```

Comment:
I don't think this mapping "ProxyStatuses -> string" makes sense, a mapping of a primitive should be justify when it's used always as an abstract type and it can be changed for a composed type in the future without changing the APIs.

A clear example is like a PrimaryKey mapping or similar, but it's not the case on this so I suggest to not add that abstraction.

Also in the UI this is mapped directly with the raw type

https://github.com/kiali/kiali-ui/pull/1884/files?file-filters%5B%5D=.ts#diff-0e0ca9c6860e3c885bf7cb495a0ad59bR78

So, I don't see why CDS and other members can be just `string`.